Workbooks/WaasUpdateInsights/DeliveryOptimization/DeliveryOptimization.workbook

```diff
@@ -0,0 +1,366 @@
+{
+  "version": "Notebook/1.0",
+  "items": [
+    {
+      "type": 9,
+      "content": {
+        "version": "KqlParameterItem/1.0",
+        "parameters": [
+          {
+            "id": "e0d2d74e-bd88-4208-bb51-10a5491a47bb",
+            "version": "KqlParameterItem/1.0",
+            "name": "_SnapshotTime",
+            "label": "Snapshot Time",
+            "type": 1,
+            "description": "Time indicating the latest snapshot",
+            "isRequired": true,
+            "query": "WUDOAggregatedStatus | project TimeGenerated\r\n| union WaaSUpdateStatus | project  TimeGenerated \r\n| where TimeGenerated >= ago(28d)\r\n| summarize max(TimeGenerated)",
+            "isHiddenWhenLocked": true,
+            "timeContext": {
+              "durationMs": 86400000
```

Comment:
the query above has >=ago 28d but right here it looks like the time range for this query is fixed to last 24h?  if this is truly supposed to be 28d then you'd have to set this query's time range to "set in query" in the parameter ux.  Also, if that is true, then this could also be a very expensive initial query and this query has to finish before anything else can start since all the other queries depend on it? (do these tables possibly have lots of rows?  most workspaces have their hot cache set to 14d so anything outside that could be really expensive?)

i don't know how any of your things specifically work, this is just a general comment when i see large time ranges by default in queries.  customers with huge amounts of data in tables could have a really bad time?

Reply:
and conversely, if there's NO data in the tables (one of the workspaces i tried it on, then you get this:

![image](https://user-images.githubusercontent.com/10158007/145654390-adbb0b82-7f51-408d-ad5d-d76d8d5b7d17.png)


the query returns no rows, so this required param has no value, and the rest of the workbook looks like this:
![image](https://user-images.githubusercontent.com/10158007/145654428-70e19d59-e65e-425f-ae03-5fcdce4e1944.png)


where it tells users to set a parameter they can't see, and can't fix?

Reply:
We updated the query to look for 7 days and the time windows is now 7 days as well. 

We don't need the customer to set the Snapshot Time. Ideally we should have data within 7 days, if not, it is beyond our SLA and we'd be working on fixing such an issue. This is consistent with other workbooks in the WaasInsights solution.